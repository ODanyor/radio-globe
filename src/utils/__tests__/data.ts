import dataUtils from 'utils/data';

describe('utils data:', () => {
  it('getItemId', () => {
    jest.spyOn(dataUtils, 'getItemId');
    const mockedGetItemId = dataUtils.getItemId as jest.Mock;
    const mockItems = [
      {
        href: '/listen/wit-80-s/mockItemId',
      },
      {
        url: '/visit/lyon/mockItemId',
      },
      {
        page: {
          url: '/visit/lyon/mockItemId',
        },
      },
    ];
    
    mockItems.forEach(mockItem => mockedGetItemId(mockItem));
    
    expect(mockedGetItemId).toBeCalledTimes(mockItems.length);
    expect(mockedGetItemId.mock.results).toMatchSnapshot();
  });
});
