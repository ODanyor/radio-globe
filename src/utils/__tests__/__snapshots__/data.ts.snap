// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils data: getItemId 1`] = `
Array [
  Object {
    "type": "return",
    "value": "mockItemId",
  },
  Object {
    "type": "return",
    "value": "mockItemId",
  },
  Object {
    "type": "return",
    "value": "mockItemId",
  },
]
`;
